monitor.cpp

```diff
@@ -1,38 +1,38 @@
 #include "./monitor.h"
-#include <assert.h>
+#include <cassert>
 #include <thread>
 #include <chrono>
 #include <iostream>
+#include <vector>
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
+VitalRange vitalRanges[] = {
+    {Temperature, 95.0f, 102.0f, "Temperature is critical!\n"},
+    {PulseRate, 60.0f, 100.0f, "Pulse Rate is out of range!\n"},
+    {SPO2, 90.0f, std::numeric_limits<float>::max(), "Oxygen Saturation out of range!\n"}
+};
+
+void showMessage(std::string message) {
+    cout << message;
     for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
+    cout << "\r* " << flush;
+    sleep_for(seconds(1));
+    cout << "\r *" << flush;
+    sleep_for(seconds(1));
   }
-  return 1;
+}
+
+bool isVitalInRange(const VitalValue& vitalValue, const VitalRange& vitalRange) {
+    return vitalValue.value >= vitalRange.min && vitalValue.value <= vitalRange.max;
+}
+
+int vitalsOk(std::vector<VitalValue> inputVitals) {
```

Comment:
This function's return is a Boolean, indicating if the vitals are ok. Can be `bool` instead of `int`?